src/main/java/com/flightstats/hub/metrics/JVMMetricsService.java

```diff
@@ -0,0 +1,57 @@
+package com.flightstats.hub.metrics;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
+import com.codahale.metrics.jvm.CachedThreadStatesGaugeSet;
+import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
+import com.codahale.metrics.ScheduledReporter;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
+import java.util.concurrent.TimeUnit;
+
+@Singleton
+public class JVMMetricsService extends AbstractIdleService {
```

Comment:
IMHO a better name for this would be something like "MetricsLifecycle", as it's not really "about" JVM metrics, it's more about setting up the Metrics Registry, the publisher, and the registering the shutdown hooks